docs/features/task-types.md

```diff
@@ -0,0 +1,90 @@
+Async Task Types in C#
+======================
+Extend `async` to support any _task type_ that matches a specific pattern, not just the well known types
+`Sytem.Threading.Tasks.Task` and `Sytem.Threading.Tasks.Task<T>`.
+
+## Task Type
+A _task type_ is a `class` or `struct` with an associated _builder type_ identified
+with `System.Runtime.CompilerServices.AsyncMethodBuilderAttribute`.
+
+A _task type_ can be non-generic, for async methods that do not return a value, or generic, for methods that return a value.
+Generic _task types_ have a `Result` property.
```

Comment:
Thanks for catching this. In fact, there is no requirement for a `Result` property on the task type after all. Instead, the requirement is on the awaiter type.